src/api/java/com/minecolonies/api/colony/buildings/modules/IBuildingModule.java

```diff
@@ -0,0 +1,109 @@
+package com.minecolonies.api.colony.buildings.modules;
+
+import com.minecolonies.api.colony.IColony;
+import com.minecolonies.api.colony.buildings.IBuilding;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.nbt.CompoundNBT;
+import net.minecraft.network.PacketBuffer;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Interface describing the different modules.
+ */
+public interface IBuildingModule
```

Comment:
Make this extend `NBTSerializable` or whatever the name of the vanilla interface is, and do not declare your own custom methods.

Reply:
The method signature we use is slightly different than NBTSerializable, so not possible